### PR TITLE
[feat] - add new etcd-disk flavor

### PR DESCRIPTION
**Pull request by AshleyDumaine**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**: This creates a new flavor to configure etcd to be on a separate disk

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: templates/flavors/default/kubeadmControlPlane.yaml [hunk 1]
```diff
@@ -36,25 +36,12 @@ spec:
     preKubeadmCommands:
       - /kubeadm-pre-init.sh '{{ ds.meta_data.label }}' "${KUBERNETES_VERSION}"
     clusterConfiguration:
-      etcd:
```
Comment (eljohnson92) [inline templates/flavors/default/kubeadmControlPlane.yaml:39]: do we want to keep this even if we don't have separate disks?

Comment (AshleyDumaine) [reply templates/flavors/default/kubeadmControlPlane.yaml:39]: Yeah I think that sounds good to me, I'll add a note in the etcd docs

File: templates/flavors/etcd-disk/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      metadata:
+        name: ${CLUSTER_NAME}-control-plane
+      spec:
+        kubeadmConfigSpec:
+          clusterConfiguration:
+            etcd:
+              local:
+                dataDir: /var/lib/etcd_data/etcd
+                extraArgs:
+                  quota-backend-bytes: "8589934592"
```
Comment (eljohnson92) [inline templates/flavors/etcd-disk/kustomization.yaml:40]: do we still need this section if it is in the default flavor?

Comment (AshleyDumaine) [reply templates/flavors/etcd-disk/kustomization.yaml:40]: We technically don't but I wanted to leave it in for clarity. I'm fine with removing it

**Review by eljohnson92 — APPROVED** [review]

**Pull request merged**